File: array/groupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        from collections import defaultdict 
        res = defaultdict(list)
        for item in strs:
            t = [0 for i in range(26)]
            for i in item:
                t[ord('a')-ord(i)]+=1
            print(t, item)
            res[tuple(t)].append(item)
        return res.values()


        

strs = ["eat","tea","tan","ate","nat","bat"]
Solution().groupAnagrams(strs)
        

File: array/groupAnagrams.go
package main

import "fmt"

func groupAnagrams(strs []string) [][]string {
	res := [][]string{}
	m := map[[26]int][]string{}

	for _, str := range strs {
		t := [26]int{}
		for _, s := range str {
			t[s-'a'] += 1
		}
		m[t] = append(m[t], str)

		fmt.Println(t, str)
	}
	for _, val := range m {
		res = append(res, val)

	}
	return res
}
func main() {
	groupAnagrams([]string{"eat", "tea", "tan", "ate", "nat", "bat"})

}
